### DISCUSSION TITLE
Lengthening of discord identifiers (18 to 19 characters)

### DISCUSSION THREAD

**Discussion by OcelusPRO** [discord-jda/JDA#1958] [Questions and Help] [answered]
Hey,

Some time ago (years) the identifiers on discord went from 17 to 18 characters, it seems that we are slowly approaching a new change from 18 to 19 characters (I don't know when exactly)

So my question is, is JDA ready for this change when it happens?

**Comment by MinnDevelopment** [comment, marked as answer]
JDA handles IDs as longs and thus does not care about the number of digits.

**Reply by OcelusPRO** [reply]
The concern is especially for the Message#getMentionedX() methods which will be based on the content of the msg and could therefore be affected by the length,
I did not find how these methods handle the recovery in the message that's why I went to the questions

**Reply by MinnDevelopment** [reply]
Mentions are provided by the API as user objects. We handle the IDs like any other ID in the API as a long.
